database/interface.go

```diff
@@ -65,6 +65,8 @@ type Iterator interface {
 	iterator.IteratorSeeker
 	Key() []byte
 	Value() []byte
+	Release()
```

Comment:
I don't think you need to change the interface at this level, see the existing https://godoc.org/github.com/syndtr/goleveldb/leveldb/iterator#CommonIterator

Reply:
i.e., maybe replace `iterator.IteratorSeeker` with `iterator.CommonIterator`?